server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesCreationProcessor.java

```diff
@@ -134,6 +135,9 @@ private void createMailbox(MailboxCreationId mailboxCreationId, MailboxCreateReq
 
     private void ensureValidMailboxName(MailboxCreateRequest mailboxRequest, MailboxSession mailboxSession) {
         String name = mailboxRequest.getName();
+        if (name.length() >= MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE) {
+            throw new MailboxNameException(String.format("The mailbox name length '%s' is over limitation: %s", name, MailboxConstants.DEFAULT_LIMIT_MAILBOX_NAME_SIZE));
+        }
```

Comment:
In fact I tried but it made more complicated for manager layer when it contains so more user's businesses. So as my point of view, we validate them from protocol layer.
But it depends our architecture so let me know if we need to move those validation to manager layer.